src/java/org/apache/cassandra/cql3/UntypedResultSet.java

```diff
@@ -351,7 +351,11 @@ public ByteBuffer getBlob(String column)
 
         public String getString(String column)
         {
-            return UTF8Type.instance.compose(data.get(column));
```

Comment:
I modify all other comment, but for this ,I should left it unchangede. When we upgrade from previous version that have not got the new added compaction_type column , the data.get(column) will return null ,and so a npe will appears, we can see the getInt and getUUID method below ,they have consider the case of null value.


Firstly , I considered changing all the methods of the corresponding type of get method  to consider the case where value is null, but I gave up later. I thought it would be more appropriate to open a new jira.

Reply:
You can use `public boolean has(String column)` in that class to check if it is null and if it is not then you call `getString` otherwise you return null. It is not necessary to change this method.

It is ok to open new ticket to investigate if we should not solve this in more robust manner but that is not objective of this ticket.